internal/bus/message.go

```diff
@@ -0,0 +1,62 @@
+package bus
+
+import (
+	"github.com/segmentio/kafka-go"
+)
+
+// GenericMessage is a platform independent message.
+type GenericMessage struct {
+	// Topic of the message. Some producers already have associated topic, in that case Topic from the message will be ignored.
+	Topic string
+
+	// Key is used for topic partitioning. Can be nil.
+	Key []byte
+
+	// Value is the payload. Typically, a JSON marshaled data.
+	Value []byte
+
+	// List of key-value pairs for each message.
+	Headers map[string]string
+}
```

Comment:
This is the main message struct that is used in our own bus interface. It can be converted (copyied) to/from Kafka native message, which is exactly the same (plus few more fields).